### DISCUSSION TITLE
Why most `Iced` examples have public `main` function?

### DISCUSSION THREAD

**Discussion by lucatrv** [hecrj/iced#1818] [Q&A] [answered]
As in subject, why use `pub fn main` instead of `fn main`?

See for instance documentation examples for [`Application`](https://docs.rs/iced/latest/iced/application/trait.Application.html#a-simple-hello-world) and [`Sandbox`](https://docs.rs/iced/latest/iced/trait.Sandbox.html#a-simple-hello-world). Moreover, most provided code examples use `pub fn main` (for instance [Arc](https://github.com/iced-rs/iced/tree/master/examples/arc)), but some use `fn main` (for instance [Sierpinski Triangle Emulator](https://github.com/iced-rs/iced/tree/master/examples/sierpinski_triangle)).

According to this [discussion](https://users.rust-lang.org/t/what-does-pub-main-do/56005), `pub fn main` is needed only if you also build a library from the same source file, which is a rare situation, so I would expect `Iced` examples to employ the most common use case `fn main`.

**Comment by hecrj** [comment, marked as answer]
I personally like to explicitly declare it public since it is the main entrypoint of an application and, therefore, being called externally. The examples that don't use it are most likely contributions by other users.

**Comment by lucatrv** [comment]
I see... Just to give my 2 cents on this, IMHO in Rust it is good practice to declare public only the very bare minimum that needs to be accessed externally. As an example, by design even if you declare a `struct` as public, then you also need to declare every element as public to have it accessible. Actually in Rust [everything is private by default](https://doc.rust-lang.org/reference/visibility-and-privacy.html) to improve safety. By declaring `main` as public, you allow everyone to run it as if it was a library function, which is not advisable unless it is specifically designed for that. That is why IMHO in Rust it would be better not to declare `main` as public in general.

**Reply by hecrj** [reply]
You did not directly address my point.

The situation you mention is very rare and harmless (`main` functions are inherently public, self-contained, and standalone). I feel your argument doesn't do much to invalidate my point (`main` functions are inherently public!).

I also feel this is a very nitpicky discussion. Why should we waste any energy on this?

**Comment by lucatrv** [comment]
OK never mind, my question was for me to understand why some `main` functions were marked as public, and you already answered that. So I am closing this discussion and I agree we should not waste energy on this.
